rclcpp/src/rclcpp/exceptions.cpp

```diff
@@ -73,6 +73,40 @@ throw_from_rcl_error(
   }
 }
 
+  std::exception_ptr
```

Comment:
Oops, when I wrote #647 I didn't realize returning a base exception type only triggers the copy constructor of the base type, and the derived exception type would be lost. Returning a pointer makes sense to me now.